Update default sizes for new dash width

kdoh opened the pull request:
Updates default sizes when adding cards to the dashboard to reflect the updated dashboard width.

- Completes Milestone 2: https://github.com/metabase/metabase/issues/36358

adam-james-v committed:
Dashboard PUT api endpoint accepts width changes and updates appdb

update-dashboard function now is aware of the :width key so those changes can end up in the transaction.

Also added a width test that asserts that the value's default is "fixed", it can be changed, eg. to "full", but cannot
be changed to other values.

adam-james-v committed:
Migration adding 'width' to Dashboards

3 migrations:
 - 1st adding the width column with default value of 'fixed'
 - 2nd updating all existing dashboards to have width 'full', which corresponds to what the current behaviour is (will
 be the 'old' behaviour after the fixed-width project lands).
   - The rolloback here is necessary but we don't care what happens as the column will be dropped immediately in the
   next rollback anyway
 - 3rd sets the notNullableConstraint. DefaultNull is 'full' here, just in case there's an existing dashboard whose
 width value is not yet set from the 1st migration. Don't know how that could happen, but its here in case

adam-james-v committed:
Add width to revision tests

adam-james-v committed:
Fix dashboard revision tests.

:width key is now needed in some revision tests. As well we need a string communicating that the :width setting has
changed from 'full' to 'fixed' or vice-versa.

adam-james-v committed:
Fix comments/remarks in migration to be accurate

adam-james-v committed:
Merge branch 'master' into fixed-width-dashboard-migration

adam-james-v committed:
Attempt to fix default not working mysql/mariadb

adam-james-v committed:
Merge branch 'fixed-width-dashboard-migration' of https://github.com/metabase/metabase into fixed-width-dashboard-migration

adam-james-v committed:
Set default in dashboard model

Signed-off-by: Adam James <adam.vermeer2@gmail.com>

adam-james-v committed:
Revert default :width 'fixed' value.

JesseSDevaney committed:
dashboard fixed width FE implementation

JesseSDevaney committed:
adjust popover shadow styling as the popover was blending in with the header

adam-james-v committed:
Explicitly add default value 'fixed' for MySQL/MariaDB

JesseSDevaney committed:
adjust extra button popover offset

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
add E2E to validate behavior

JesseSDevaney committed:
add fixed width container to public dashboards

JesseSDevaney committed:
fix public embedding not respecting dashboard width setting

JesseSDevaney committed:
add test for public dashboards

adam-james-v committed:
Fix embed test failures

JesseSDevaney committed:
add fixed width to x-ray dashboards

JesseSDevaney committed:
reduce code duplication

JesseSDevaney committed:
move FixedWidthContainer into DashboardGridConnected

- We get the fixed-width w/o code duplication across AutomaticDashboardApp, PublicDashboard, and Dashboard

JesseSDevaney committed:
update E2E tests to reflect actual intended behavior

JesseSDevaney committed:
fix type errors

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

kdoh committed:
tweak default width of cards to accomodate fixed width

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
adjust tooltip button after merging changes from master

JesseSDevaney committed:
fix bar chart test failure

JesseSDevaney committed:
fix dashboard filters date test failure

- new ellipsis button in dashboard edit mode broke the test

JesseSDevaney committed:
adjust test for new fixed width dashboards

JesseSDevaney committed:
adjust test for new fixed width dashboards

cdeweyx commented on frontend/src/metabase/visualizations/shared/utils/sizes.ts:
```diff
@@ -43,23 +43,23 @@ const VISUALIZATION_SIZES: {
   },
   funnel: {
     min: { width: 4, height: 3 },
     default: { width: 8, height: 6 },
```
Any reason why funnels are still 8 default? That's the only one here that I was wondering, why not bump that up to 12 as well.

kdoh replied:
Nope, just forgot. Thanks for the reminder.

cdeweyx reviewed (approved):
Left one question on funnels but overall much much better

JesseSDevaney committed:
adjust test for new fixed width dashboards

kdoh committed:
bump funnel default width

JesseSDevaney committed:
adjust test for new fixed width dashboards

JesseSDevaney committed:
Fix double overlay for LineAreaBarChart dash-cards

- No longer has an overlay that persists from editing mode to  viewing mode
- Adjust click-behavior tests to account for new grid-width. Old test relied on a dashcard being taller than they will now default to being. i.e. Move the chosen point/row to down an index.

JesseSDevaney committed:
Merge FixedWidthContainer with dashboard grid div

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
Merge branch 'feature-36358/default-dashboards-to-a-fixed-width' into dash-width-default-sizes

JesseSDevaney commented:
Waiting to merge until Milestone 1 is merged
- https://github.com/metabase/metabase/pull/38377

alxnddr committed:
supress brush events while editing timeseries line charts on dashboard

JesseSDevaney committed:
fix failing test

- test failed because I did not update this assertion based on previous changes

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
fix broken link input in editing mode

JesseSDevaney committed:
Merge branch 'feature-36358/default-dashboards-to-a-fixed-width' into dash-width-default-sizes

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
fix pointer events issue on text cards in edit mode

JesseSDevaney committed:
Merge branch 'master' into feature-36358/default-dashboards-to-a-fixed-width

JesseSDevaney committed:
Merge branch 'feature-36358/default-dashboards-to-a-fixed-width' into dash-width-default-sizes

JesseSDevaney committed:
Merge branch 'master' into dash-width-default-sizes

JesseSDevaney committed:
update tests for new dash-card default sizes

JesseSDevaney committed:
update tabs card moving test because of new default card sizes

JesseSDevaney committed:
Merge branch 'master' into dash-width-default-sizes

kulyk commented on frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx:
```diff
@@ -413,7 +413,10 @@ class DashboardHeaderContainer extends Component<DashboardHeaderProps> {
       buttons.push(
         <DashboardButtonTooltip key="add-link-card" label={t`Add link card`}>
-          <DashboardHeaderButton onClick={() => this.onAddLinkCard()}>
+          <DashboardHeaderButton
+            aria-label={t`Add link card`}
+            onClick={() => this.onAddLinkCard()}
+          >
```
❤️

perivamsi commented:
force merging it because of a failing unit test that is [unrelated to this change](https://github.com/metabase/metabase/pull/38433)
